src/backend/utils/adt/date.c

```diff
@@ -1384,6 +1386,9 @@ time_in(PG_FUNCTION_ARGS)
 	int			dtype;
 	int			ftype[MAXDATEFIELDS];
 
+	if (tsql_time_in_hook && sql_dialect == SQL_DIALECT_TSQL)
```

Comment:
we can move dialect check inside the hook itself and pass result as out parameter